src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -2586,42 +2596,557 @@ class C
+                );
+        }
 
-            // PROTOTYPE(tuples) uncomment above once binding is fixed
-            var comp = CreateCompilationWithMscorlib(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef }, parseOptions: TestOptions.Regular.WithTuplesFeature());
+        [Fact(Skip = "PROTOTYPE(tuples)")]
+        public void InferTypeOfTypelessDeclaration()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        (var (x1, x2), string x3) = ((1, 2), null);
+        System.Console.WriteLine(x1 + "" "" + x2 + "" "" + x3);
+    }
+}
+";
+            // crash
+            var comp = CompileAndVerify(source, expectedOutput: "1 2 ", parseOptions: TestOptions.Regular);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void VarVarDeclaration()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        (var (x1, x2), var x3) = Pair.Create(Pair.Create(1, ""hello""), 2);
+        System.Console.WriteLine(x1 + "" "" + x2 + "" "" + x3);
+    }
+}
+" + commonSource;
+
+            string expected =
+@"Deconstructing ((1, hello), 2)
+Deconstructing (1, hello)
+1 hello 2";
+
+            Action<ModuleSymbol> validator = (ModuleSymbol module) =>
+            {
+                var sourceModule = (SourceModuleSymbol)module;
+                var compilation = sourceModule.DeclaringCompilation;
+                var tree = compilation.SyntaxTrees.First();
+                var model = compilation.GetSemanticModel(tree);
+
+                var x1 = GetDeconstructionLocal(tree, "x1");
+                var x1Ref = GetReference(tree, "x1");
+                VerifyModelForDeconstructionLocal(model, x1, x1Ref);
+
+                var x2 = GetDeconstructionLocal(tree, "x2");
+                var x2Ref = GetReference(tree, "x2");
+                VerifyModelForDeconstructionLocal(model, x2, x2Ref);
+
+                var x3 = GetDeconstructionLocal(tree, "x3");
+                var x3Ref = GetReference(tree, "x3");
+                VerifyModelForDeconstructionLocal(model, x3, x3Ref);
+            };
+
+            var comp = CompileAndVerify(source, expectedOutput: expected, parseOptions: TestOptions.Regular, sourceSymbolValidator: validator);
+            comp.VerifyDiagnostics();
+        }
+
+        private static void VerifyModelForDeconstructionLocal(SemanticModel model, VariableDeclarationSyntax decl, params IdentifierNameSyntax[] references)
+        {
+            var variableDeclaratorSyntax = decl.Variables.Single();
+            var symbol = model.GetDeclaredSymbol(variableDeclaratorSyntax);
+            Assert.Equal(variableDeclaratorSyntax.Identifier.ValueText, symbol.Name);
+            Assert.Equal(LocalDeclarationKind.RegularVariable, ((LocalSymbol)symbol).DeclarationKind);
+            Assert.Same(symbol, model.LookupSymbols(decl.SpanStart, name: variableDeclaratorSyntax.Identifier.ValueText).Single());
```

Comment:
I believe it calls a different overload of GetDeclaraedSymbol.

---
In reply to: [68856288](https://github.com/dotnet/roslyn/pull/12224#discussion_r68856288) [](ancestors = 68856288)